cmd/gitbase/command/server.go

```diff
@@ -318,18 +319,34 @@ func (c *Server) addDirectories() error {
 
 func (c *Server) addDirectory(d directory) error {
 	if d.Format == "siva" {
-		sivaOpts := siva.LibraryOptions{
-			Transactional: true,
-			RootedRepo:    d.Rooted,
-			Cache:         c.sharedCache,
-			Bucket:        d.Bucket,
-			Performance:   true,
-			RegistryCache: 100000,
-		}
+		var lib borges.Library
+		var err error
+
+		if d.Rooted {
+			sivaOpts := &siva.LibraryOptions{
+				Transactional: true,
+				RootedRepo:    d.Rooted,
+				Cache:         c.sharedCache,
+				Bucket:        d.Bucket,
+				Performance:   true,
+				RegistryCache: 100000,
+			}
 
-		lib, err := siva.NewLibrary(d.Path, osfs.New(d.Path), sivaOpts)
-		if err != nil {
-			return err
+			lib, err = siva.NewLibrary(d.Path, osfs.New(d.Path), sivaOpts)
```

Comment:
so maybe we can combine these 2 libs, e.g.:
```go
lib := library.New()
//....
lib = library.WithSiva(lib, sivaOpts)
// or lib = lib.WithSiva(sivaOpts)
// or lib = lib.WithSiva(...).WithOldSiva(...)
```

Reply:
Right now there are 4 types of libraries:

* `plain`: holds normal repositores
* `siva`: repositories in siva files, the new format
* `oldsiva`: sivas generated by borges, old format
* `libraries`: a meta library that can hold any other library.

Here we are generating libraries of different types depending on the options we get and adding it to the "meta library" `rootLibrary`. We could have added an `AddSiva` or `AddOldsiva` to the `libraries.Library` structure but it's meant to hold any structure that implements the interface `borges.Library`. I don't think that coupling `libraries.Library` with the  already implemented types of library adds anything.

Reply:
This is just my taste, so no worries. In this case I thought we may have `WithOptions(...)`
and in siva, plain, ... you have have certain type's options, so library would be top level, and all types would define specific options.